presto-orc/src/main/java/com/facebook/presto/orc/Filters.java

```diff
@@ -714,4 +747,266 @@ public static Filter createMultiRange(List<Filter> filters, boolean nullAllowed)
             return new MultiRange(filters, nullAllowed);
         }
     }
+
+    public static class PositionalFilter
+            extends Filter
+    {
+        // The set of rows for which this specifies a Filter.
+        private int[] filterRows;
```

Comment:
There are a lot of variables in this class and I'm confused by their names. Based on my understanding of the functionality, I suggest to rename these as follows. I also suggest to  separate groups of related variables with empty lines.

```
        // Row numbers in ascending order
        private int[] positions;
        // Rows numbers in the containing structure (map or list)
        private int[] inputNumbers;
        // Number of valid entries in positions (and inputNumbers)
        private int numPositions;
        // Filters for individual rows. Null entry means no filter.
        private Filter[] filters;

        // True if all rows will be evaluated in sequence.
        private boolean scanAllPositions;
        // A subset of rows that will be evaluated in sequence defined as
        // indexes into positions array in ascending order.
        private int[] selectedPositionIndexes;
        // Number of valid entries in the selectedPositionIndexes.
        private int numSelectedPositions;

        // Index of the next filter to apply.
        private int nextFilterIndex;

        // Count of upcoming textXxx calls to fail. Suppose an array of
        // 4 elements with a failed filter at first element. There
        // would be 3 elements to go that are in any case
        // disqualified, so the failing filter on the first element
        // would set this to 3.
        private int numNextPositionsToFail;

        private final StructFilter source;
```